core/workspace.js

```diff
@@ -84,6 +84,18 @@ Blockly.Workspace = function(opt_options) {
    * @private
    */
   this.variableMap_ = new Blockly.VariableMap(this);
+
+  /**
+   * Blocks in the flyout can refer to variables that don't exist in the
+   * workspace.  For instance, the "get item in list" block refers to an "item"
+   * variable regardless of whether the variable has been created yet.
+   * A FieldVariable must always refer to a Blockly.VariableModel.  We reconcile
+   * these by tracking "potential" variables in the flyout.  These variables
+   * become real when references to them are dragged into the main workspace.
+   * @type {!Blockly.VariableMap}
+   * @private
+   */
+  this.potentialVariableMap_ = new Blockly.VariableMap(this);
```

Comment:
This doesn't feel right. It implies the flyout workspaces (and mutator workspaces?) also have their own independent potential variables maps.

This might be something that should be null by default, and set during inject(..), assigning the flyout workspace's variable map as the potential variable map of the main workspace.

Reply:
Fixed.